pandas/tseries/period.py

```diff
@@ -811,6 +811,16 @@ def _format_native_types(self, na_rep=u('NaT'), date_format=None, **kwargs):
         values[imask] = np.array([formatter(dt) for dt in values[imask]])
         return values
 
+    def putmask(self, mask, value):
+        """
+        return a new Index of the values set with the mask
+
+        See also
+        --------
```

Comment:
another thought.

instead of calling it ``_assert_can_do_op`` if you made it:

``value = _convert_for_op(value)`` then  you just need to define this for ``period.py`` as

```
def _convert_for_op(self, value):
    return Period(value)
```

(you already do the conversions for ``tseries/index.py``). 

Then you don't need the putmask here either